### ISSUE TITLE
2 Handed Block Break Delay

### ISSUE THREAD

**Issue by jpv776** [ferriarnus/VivecraftForge#326]
Using a Forge 1.21.1 server and breaking blocks with 2 hands causes blocks to take longer to break/drop. Same thing also happens when breaking a block one handed very fast.

**Comment by fayer3** [comment]
take longer compared to what?
singleplayer, or older versions

**Comment by jpv776** [comment]
> take longer compared to what? singleplayer, or older versions

I haven't tried it in singleplayer but to give more clarification I break a block with both my actual hands, not by holding the trigger and it sits there for about 3 seconds and then disappears and drops the block on the ground.

**Comment by fayer3** [comment]
could be that they added a minumum breakage time on the server or something
can you check if the time from when you start till it drops, is roughly equal to breaking it with the trigger?

**Comment by jpv776** [comment]
I've tried the trigger and it breaks as normal and doesn't fake break like described, if breaking with one hand at a normal pace it breaks normally as well.

**Comment by fayer3** [comment]
yeah I'm more innterested of the time comparison, but I can check that myself I guess

**Comment by fayer3** [comment]
yeah, looks like that happens when breaking it faster than vanilla

**Comment by jpv776** [comment]
I can upload a comparison, trying in survival yields the same results as the server.

**Comment by fayer3** [comment]
need to dig into the code, if I can remove that, maybe not possible

**Comment by jpv776** [comment]
https://github.com/user-attachments/assets/4684629a-1b7b-4e8f-8d33-3f5f8179a201

**Comment by fayer3** [comment]
yeah that looks like it just waits the vanilla time, before actually braking

**Comment by Techjar** [comment]
Probably vanilla's server-side anti-cheat checks.
